source/distributed/tria.cc

```diff
@@ -1124,10 +1124,545 @@ namespace
+        quad_cell_relations.size());
+
+      // Iterate over all cells, call all callback functions on each cell,
+      // and store their data in the corresponding buffer scope.
+      {
+        auto cit_qcrel    = quad_cell_relations.cbegin();
+        auto it_data_cell = packed_data.begin();
+        for (; cit_qcrel != quad_cell_relations.cend();
+             ++cit_qcrel, ++it_data_cell)
+          {
+            const auto &cell_status = std::get<1>(*cit_qcrel);
+            const auto &dealii_cell = std::get<2>(*cit_qcrel);
+
+#  ifdef DEBUG
+            // Assertions about the tree structure.
+            switch (cell_status)
+              {
+                case parallel::distributed::Triangulation<dim, spacedim>::
+                  CELL_PERSIST:
+                case parallel::distributed::Triangulation<dim, spacedim>::
+                  CELL_REFINE:
+                  // double check the condition that we will only ever attach
+                  // data to active cells when we get here
+                  Assert(dealii_cell->active(), ExcInternalError());
+                  break;
+
+                case parallel::distributed::Triangulation<dim, spacedim>::
+                  CELL_COARSEN:
+                  // double check the condition that we will only ever attach
+                  // data to cells with children when we get here. however, we
+                  // can only tolerate one level of coarsening at a time, so
+                  // check that the children are all active
+                  Assert(dealii_cell->active() == false, ExcInternalError());
+                  for (unsigned int c = 0;
+                       c < GeometryInfo<dim>::max_children_per_cell;
+                       ++c)
+                    Assert(dealii_cell->child(c)->active(), ExcInternalError());
+                  break;
+
+                case parallel::distributed::Triangulation<dim, spacedim>::
+                  CELL_INVALID:
+                  // do nothing on invalid cells
+                  break;
+
+                default:
+                  Assert(false, ExcInternalError());
+                  break;
+              }
+#  endif
+
+            // Reserve memory corresponding to the amount of callback
+            // function that will be called.
+            // On cells flagged with CELL_INVALID, only its CellStatus
+            // will be stored.
+            const unsigned int num_callbacks_on_cell =
+              1 +
+              ((cell_status ==
+                parallel::distributed::Triangulation<dim,
+                                                     spacedim>::CELL_INVALID) ?
+                 0 :
+                 pack_callbacks.size());
+            it_data_cell->resize(num_callbacks_on_cell);
+
+            // We continue with packing all data on this specific cell.
+            // First, we pack the CellStatus information.
+            auto it_data_callback = it_data_cell->begin();
```

Comment:
same here -- make `it` a *postfix*, not a *prefix*

Reply:
also, the iterator is not into the callbacks, but in fact into the data. so maybe just call it `data_ptr`.